### PR TITLE
#[UIC-719] Non root url related support

### PR DESCRIPTION
**Pull request by ankursinghal2005**
[UIC-719](https://guavus-jira.atlassian.net/browse/UIC-719)

### CATEGORY

Choose one

- [ ] Bug Fix
- [X] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Add functionality to append common prefix as ‘xyz’ before each request of superset (non root deployment). Adding non root prefix is optional.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [X] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by ankursinghal2005** [comment]
Remaining Improvement in PR -:

1. Convert $.ajax http call to SupersetClient
2. SupersetClient.getUrl method should be in common palace and refactor needed

File: superset/__init__.py [hunk 1]
```diff
@@ -192,10 +192,29 @@ def __call__(self, environ, start_response):
...
+
+       # if environ['PATH_INFO'].startswith(self.prefix):
+        environ['PATH_INFO'] = environ['PATH_INFO'][len(self.prefix):]
+
+        environ['SCRIPT_NAME'] = self.prefix
+        return self.app(environ, start_response)
+        # else:
+        #     start_response('404', [('Content-Type', 'text/plain')])
+        #     return ["This url does not belong to the app.".encode()]
+
```
[the remark this answers is not in the document]
Comment (bipinsoniguavus) [reply superset/__init__.py]: Done

**Comment by bipinsoniguavus** [comment]
> Remaining Improvement in PR -:
> 
> 1. Convert $.ajax http call to SupersetClient
> 2. SupersetClient.getUrl method should be in common place and refactor needed

Handled #1 in this [PR](https://github.com/Guavus/incubator-superset/pull/207)
`SupersetClient.get` change, we should not move now as upgrade will have bigger changes in all the files even if we are planning to contribute this to OSS, it might not get merged before we plan upgrade.

File: superset/assets/src/setup/setupClient.js [hunk 2]
```diff
@@ -17,11 +17,35 @@
...
+    }
+    console.log("Setup Client -->", endpoint, url);
+  
+    if (typeof url === 'string') return url;
+
+    const host = inputHost || this.host;
+    const cleanHost = host.slice(-1) === '/' ? host.slice(0, -1) : host; // no backslash
+
+    return `${this.protocol}//${cleanHost}/${endpoint[0] === '/' ? endpoint.slice(1) : endpoint}`;
+  }
```
Comment (jitendra-kumawat) [inline superset/assets/src/setup/setupClient.js]: we should re-use this from
superset/assets/spec/helpers/setupSupersetClient.js or define this in common way to re-use

Comment (bipinsoniguavus) [reply superset/assets/src/setup/setupClient.js]: I checked the same, but upgrade would have conflict with the same, so we should target the same list oss upgrade.

**Pull request merged**